connectivity/service/src/main/java/org/eclipse/ditto/connectivity/service/mapping/CloudEventsMapper.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) 2022 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.ditto.connectivity.service.mapping;
+
+import static java.util.Collections.singletonList;
+import static org.eclipse.ditto.json.JsonFactory.newObject;
+
+import java.io.UnsupportedEncodingException;
+import java.util.*;
+
+import com.eclipsesource.json.Json;
+import org.eclipse.ditto.base.model.common.DittoConstants;
+import org.eclipse.ditto.base.model.exceptions.DittoJsonException;
+import org.eclipse.ditto.base.model.exceptions.DittoRuntimeException;
+import org.eclipse.ditto.base.model.headers.DittoHeaders;
+import org.eclipse.ditto.connectivity.api.ExternalMessage;
+import org.eclipse.ditto.connectivity.api.ExternalMessageFactory;
+import org.eclipse.ditto.connectivity.model.ConnectivityModelFactory;
+import org.eclipse.ditto.connectivity.model.MappingContext;
+import org.eclipse.ditto.connectivity.model.MessageMappingFailedException;
+import org.eclipse.ditto.internal.utils.akka.logging.DittoLogger;
+import org.eclipse.ditto.internal.utils.akka.logging.DittoLoggerFactory;
+import org.eclipse.ditto.json.*;
+import org.eclipse.ditto.messages.model.MessageHeaders;
+import org.eclipse.ditto.protocol.Adaptable;
+import org.eclipse.ditto.protocol.JsonifiableAdaptable;
+import org.eclipse.ditto.protocol.ProtocolFactory;
+
+/**
+ * A message mapper implementation for the Mapping incoming CloudEvents to Ditto Protocol.
+ */
+@PayloadMapper(alias = {"CloudEvents",
+    // legacy full qualified name
+    "org.eclipse.ditto.connectivity.service.mapping.CloudEventsMapper"})
+public final class CloudEventsMapper extends AbstractMessageMapper {
+
+
+  static final String CE_ID = "ce-id";
+  static final String CE_TYPE = "ce-type";
+  static final String CE_SOURCE = "ce-source";
+  static final String CE_SPECVERSION = "ce-specversion";
+
+  static final String contentType = "application/cloudevents+json";
+
+  // make all of them static
+  static final String SPECVERSION = "specversion";
+  static final String ID = "id";
+  static final String SOURCE = "source";
+  static final String TYPE = "type";
+  // declare uuid in the function
+  // make static variables uppercase
+
+
+  static final String OUTBOUNDTYPE = "org.eclipse.ditto.outbound";
+
+  static final String OUTBOUNDSPECVERSION = "1.0";
+
+  static final String OUTBOUNDSOURCE = "org.eclipse.ditto";
+
+
+  private String base64decoding(final String base64Message) {
+    byte[] messageByte = Base64.getDecoder().decode(base64Message);
+    String decodedString = new String(messageByte);
+    return decodedString;
```

Comment:
those 2 lines can be combined in 
```suggestion
    return new String(messageByte);
```